packages/codelabs/src/basics/lit-html.md

```diff
@@ -0,0 +1,1139 @@
+- A web browser that supports Web Components: Firefox, Safari, Chrome or any Chromium-based browser.
+
+**What you'll learn:**
+
+- Creating web components with lit-element
+- Templating with lit-html
+- Repeating templates
+- Conditional rendering
+- Handling events
+- Managing data
+- Creating child components
+
+**How it works**
+This codelab will go step by step, explaining each code change. At the bottom of each section, there is a "View final result" button where you can see the final code result you should end up with. The codelab is sequential, so results from the previous step carry on to the next step.
+
+At the bottom of each step, there is a final code snippet to make sure you're still on the right track or to help you when you get stuck.
+
+## Setup
+
+In this codelab, we will build a simple todo app, a great showcase for learning the basics of lit-html and lit-element.
+
+You can follow this codelab using anything that can display a simple HTML page. We recommend using an [online code editor like jsbin](https://jsbin.com/?html,output), but you could also create a basic html page using your IDE.
+
+To get started, create a simple webpage:
+
+```html
+<!DOCTYPE html>
+<html>
+  <body>
+    My todo app
+  </body>
+</html>
+```
+
+If you run this in the browser and see the message on the screen, you're good to go.
+
+## lit-element setup
+
+lit-element takes care of most of the boilerplate when writing components, providing a great developer experience while staying close to the browser platform. It does not require any build step to run in the browser, and it's only 7kb. This makes it an ideal lightweight choice.
+
+lit-element is written and distributed as es modules, this means we can import it using the browser's native module loader. Let's create a module script, and import LitElement from a CDN:
+
+```html
+<!DOCTYPE html>
+
+<html>
+  <body>
+    <script type="module">
+      import { LitElement } from 'https://unpkg.com/lit-element?module';
+    </script>
+  </body>
+</html>
+```
+
+Make sure you add `type="module"` to the script tag.
+
+Next, we need to define our web component. When writing a vanilla web component we would extend from `HTMLElement`. With lit-element we need to import and extend from `LitElement`, which itself is an extension of `HTMLElement`.
+
+```js
+class TodoApp extends LitElement {
+  connectedCallback() {
+    super.connectedCallback();
+    console.log('lit element connected');
+  }
+}
+
+customElements.define('todo-app', TodoApp);
+```
+
+We defined the tag name for our element as `todo-app`, now we need to add this to the HTML of our page:
+
+```html
+<todo-app></todo-app>
+```
+
+If you run this in the browser you should see `lit element connected` logged to the terminal.
+
+<aside class="notice">
+  Because LitElement also does some work in the connectedCallback, we must always call super.connectedCallback. This is a common source of confusion, so make sure to remember!
```

Comment:
```suggestion
  Because LitElement also does some work in the `connectedCallback`, we must always call `super.connectedCallback()`. This is a common source of confusion, so make sure to remember!
```